Mathlib/GroupTheory/Perm/Cycle/PossibleTypes.lean

```diff
@@ -0,0 +1,76 @@
+/-
+Copyright (c) 2023 Antoine Chambert-Loir. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Antoine Chambert-Loir
+-/
+
+import Mathlib.GroupTheory.Perm.Cycle.Concrete
+
+/-! # Possible cycle types of permutations
+
+* For `m : Multiset ℕ`, `Equiv.Perm.withCycleType_nonempty_iff m`
+  proves that there are permutations with cycleType `m` if and only if
+  its sum is at most `Fintype.card α` and its members are at least 2.
+
+-/
+
+variable (α : Type _) [DecidableEq α] [Fintype α]
+
+/-- There are permutations with cycleType `m` if and only if
+  its sum is at most `Fintype.card α` and its members are at least 2. -/
+theorem Equiv.Perm.with_cycleType_nonempty_iff {m : Multiset ℕ} :
+    Set.Nonempty {g : Equiv.Perm α | g.cycleType = m} ↔
```

Comment:
Writing this in terms of existence of `g` might be a bit simpler than going through `Set.Nonempty`.

Reply:
I can do either way, since I explicitly prove existence. 
My own idiosyncratic style wishes to avoid a quantifier. 
Also, the whole game (PR to come) is about the cardinality of that set.

Reply:
[Set.Nonempty](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Data/Set/Basic.html#Set.Nonempty) is already a quantifier, so this shouldn't make a real difference.

Reply:
My interpretation of "The property s.[Nonempty](https://leanprover-community.github.io/mathlib4_docs/Init/Prelude.html#Nonempty) expresses the fact that the set s is not empty. It should be used in theorem assumptions instead of ∃ x, x ∈ s or s ≠ ∅ as it gives access to a nice API thanks to the dot notation." is as follows:

if it has to be used as a theorem assumption, it'd rather be used as a theorem conclusion, for otherwise, the user would have to do some rewrite.

Reply:
(Since it's *equal* to ∃ x, x ∈  s, maybe no rewrite is necessary.)

Reply:
You can try, but `rw` usually usually only works up to syntactic equality (so it doesn't see even through definitional equality).

Reply:
So it seems to be better to have it as `Set.Nonempty`.